python/ray/train/tests/test_trainer.py

```diff
@@ -1178,44 +1058,50 @@ def test_resources(ray_start_4_cpus_4_gpus_4_extra, resource, num_requested):
 
 
 def test_gpu_requests(ray_start_4_cpus_4_gpus_4_extra):
+    class CudaTestBackend(TestBackend):
```

Comment:
This is because we used to have `os.environ[ENABLE_SHARE_CUDA_VISIBLE_DEVICES_ENV] = "1"` but with Ray Client this is not propagated to the `BackendExecutor`.  Open to suggestions on what the "right" thing to do here would be!